### PR TITLE
Makes site URL optional.

### PR DESCRIPTION
**Pull request by BertrandBordage**
Currently, we must set a URL on a each site. There are some cases where sites are no longer on any existing URLs, so I think it’s better to make this field optional.

Here are cases I met:
- a website created for an event, so it’s no longer on the web;
- a very nice website with lots of unique features that was removed from the web and replaced by a random Wordpress for political reasons, a few months after being published.

### CODE REVIEW

File: core/templates/core/wagtail_site_page.html [hunk 1]
```diff
@@ -28,10 +28,12 @@
...
-                    <a href="{{ self.site_url }}" title="{{ self.title}}" class="btn -small icon-text" data-analytics="Project|Link click">
-                        {% include "core/includes/icon.html" with icon="visit" color="white" class="i--small u-middle" %}
-                        <span>Visit site</span>
-                    </a>
+                    {% if self.site_url %}
+                        <a href="{{ self.site_url }}" title="{{ self.title }}" class="btn -small icon-text" data-analytics="Project|Link click">
+                            {% include "core/includes/icon.html" with icon="visit" color="white" class="i--small u-middle" %}
+                            <span>Visit site</span>
+                        </a>
+                    {% endif %}
```
Comment (thibaudcolas) [inline core/templates/core/wagtail_site_page.html]: 👍

File: core/templates/core/includes/sites.html [hunk 2]
```diff
@@ -17,10 +17,12 @@
...
-                    <a class="project__visit" href="{{ page.site_url }}" title="Project link" data-analytics="Project|Link click">
-                        <span class="u-sr-only">Project link</span>
-                        {% include "core/includes/icon.html" with icon="visit" color="grey" class="i--hover" %}
-                    </a>
+                    {% if page.site_url %}
+                        <a class="project__visit" href="{{ page.site_url }}" title="Project link" data-analytics="Project|Link click">
+                            <span class="u-sr-only">Project link</span>
+                            {% include "core/includes/icon.html" with icon="visit" color="grey" class="i--hover" %}
+                        </a>
+                    {% endif %}
```
Comment (thibaudcolas) [inline core/templates/core/includes/sites.html]: 👍

**Comment by BertrandBordage** [comment]
This is because other changes were made to these fields in the past, but no migration was done.
By the way, this migration is useless from a SQL perspective, as it changes nothing in the schema. I did it so Django’s migration system doesn’t complain about unmigrated changes in models.

**Comment by thibaudcolas** [comment]
I see – I vaguely remember making lots of help text changes, was still learning about migrations back then. I tmight be from this.

I'll try this locally and merge.

**Pull request merged**